### PR TITLE
Print file and line info, and member info, for each LLVM instruction

### PR DESCRIPTION
**Pull request by dmlloyd**

### CODE REVIEW

**Comment by dmlloyd** [comment]
Another change snuck in here which avoids certain sticky problems when invocation return value nodes get processed in certain situations.  I can separate it out, but it'd likely just be another PR right away anyway. 🙂

Either way is OK with me though.

**Review by DanHeidinga — COMMENTED** [review]
Overall, looks OK to me.  Would be nice to see how verbose the new output is though

File: plugins/llvm/src/main/java/org/qbicc/plugin/llvm/LLVMNodeVisitor.java [hunk 1]
```diff
@@ -766,11 +759,13 @@ public LLValue visit(final Void param, final InsertMember node) {
...
-        map(node.getInvoke().getTerminatedBlock());
-        // should already be registered by now
+        LLBasicBlock invokeBlock = map(node.getInvoke().getTerminatedBlock());
+        // should already be registered by now in most cases
         LLValue llValue = mappedValues.get(node);
         if (llValue == null) {
-            throw new IllegalStateException();
+            // map late
+            postMap(node.getInvoke().getTerminatedBlock(), invokeBlock);
+            llValue = mappedValues.get(node);
```
[the remark this answers is not in the document]
Comment (dmlloyd) [reply plugins/llvm/src/main/java/org/qbicc/plugin/llvm/LLVMNodeVisitor.java:768]: Yeah that's the one.

**Comment by dmlloyd** [comment]
> Overall, looks OK to me. Would be nice to see how verbose the new output is though

It's fairly compact:

```llvm
define i32 @exact.java.lang.String.length.s32.0(i8 addrspace(1)* %thr0, i8 addrspace(1)* %this0) "frame-pointer"="non-leaf" uwtable gc "statepoint-example" !dbg !41 {
B0:
  %L0 = icmp eq i8 addrspace(1)* %this0, null, !dbg !42 ; String.java:658 bci@1
  %L1 = bitcast i8 addrspace(1)* %this0 to %T.java.lang.String addrspace(1)*, !dbg !42 ; String.java:658 bci@1
  br i1 %L0, label %B1, label %B2, !dbg !42 ; String.java:658 bci@1
B1:
  call void (i8 addrspace(1)*) @exact.org.qbicc.runtime.main.VMHelpers.raiseNullPointerException.v.0(i8 addrspace(1)* %thr0) noreturn, !dbg !42
  unreachable, !dbg !42 ; String.java:658 bci@1
B2:
  %L2 = getelementptr %T.java.lang.String, %T.java.lang.String addrspace(1)* %L1, i32 0, i32 3, !dbg !42 ; member value
  %L3 = load atomic i8 addrspace(1)*, i8 addrspace(1)* addrspace(1)* %L2 unordered, align 8, !dbg !42 ; String.java:658 bci@1
  %L4 = icmp eq i8 addrspace(1)* %L3, null, !dbg !42 ; String.java:658 bci@4
  %L5 = bitcast i8 addrspace(1)* %L3 to %T.internal_array_B addrspace(1)*, !dbg !42 ; String.java:658 bci@4
  br i1 %L4, label %B3, label %B4, !dbg !42 ; String.java:658 bci@4
B3:
  call void (i8 addrspace(1)*) @exact.org.qbicc.runtime.main.VMHelpers.raiseNullPointerException.v.0(i8 addrspace(1)* %thr0) noreturn, !dbg !42
  unreachable, !dbg !42 ; String.java:658 bci@4
B4:
  %L6 = getelementptr %T.internal_array_B, %T.internal_array_B addrspace(1)* %L5, i32 0, i32 1, !dbg !42 ; member length
  %L7 = load atomic i32, i32 addrspace(1)* %L6 unordered, align 4, !dbg !42 ; String.java:658 bci@4
  br i1 %L0, label %B5, label %B6, !dbg !42 ; String.java:658 bci@6
B5:
  call void (i8 addrspace(1)*) @exact.org.qbicc.runtime.main.VMHelpers.raiseNullPointerException.v.0(i8 addrspace(1)* %thr0) noreturn, !dbg !42
  unreachable, !dbg !42 ; String.java:658 bci@6
B6:
  %L8 = call signext i8 (i8 addrspace(1)*, i8 addrspace(1)*) @exact.java.lang.String.coder.s8.0(i8 addrspace(1)* %thr0, i8 addrspace(1)* %this0), !dbg !42 ; String.java:658 bci@6
  %L9 = sext i8 %L8 to i32, !dbg !42 ; String.java:658 bci@6
  %LA = and i32 %L9, 31, !dbg !42 ; String.java:658 bci@9
  %LB = ashr i32 %L7, %LA, !dbg !42 ; String.java:658 bci@9
  ret i32 %LB, !dbg !42 ; String.java:658 bci@10
}
```

**Review by DanHeidinga — APPROVED** [review]
lgtm

**Pull request merged**